finished ListEntry

NoamGilad opened the pull request:
@asafDovNetzer

NoamGilad committed:
started working on amount per hour and ListEntry

NoamGilad committed:
finished ListEntry

NoamGilad committed:
started working with timestamps

NoamGilad committed:
time diff

NoamGilad committed:
trying to render the damn list

made a mess, i need to make a CalckTimeDiff component and split the code...

NoamGilad committed:
Update MainForm.js

NoamGilad committed:
123

asafDovNetzer commented on src/components/MainForm.js:
```diff
@@ -24,46 +26,50 @@ function MainFrom({ addEntryMainForm }) {
-      date,
-      from,
-      to,
+    const shift = {
```
this object named "shift" is not being used. erase it please

NoamGilad committed:
erased "shift"

NoamGilad committed:
error-modal & css

NoamGilad committed:
amount per hour, salary

not finished.

want to make the inputs's background color red if they are empty

to make the calcTimeDiff function work (Salary.js)

to make the amount per hour permanent
for now

NoamGilad committed:
...

NoamGilad committed:
started calc time diff

NoamGilad committed:
shift duration + login (dummy for now)

made MainHeader component

NoamGilad committed:
useReducer for Login

NoamGilad committed:
Nav, shitTime, useContext

NoamGilad committed:
Update Navigation.js

NoamGilad committed:
Update Navigation.js

NoamGilad committed:
נסיון לעשות סדר בבאלגן

asafDovNetzer commented on src/App.js:
```diff
@@ -1,33 +1,53 @@
+    <Fragment>
+      {!AuthCtx.isLoggedIn && (
+        <Fragment>
+          <MainHeader onLogout={AuthCtx.logoutHandler} />
```
no need to write <mainHeader /> twice. think of a way to only write it once

asafDovNetzer commented on src/components/UI/MainHeader.js:
```diff
@@ -0,0 +1,13 @@
+  return (
+    <header className={classes["main-header"]}>
+      <h1>Workings hours</h1>
+      <Navigation onLogout={props.onLogout} />
```
no need to give navigation this props since it is not used. the navigation uses context

asafDovNetzer commented on src/components/UI/MainHeader.module.css:
```diff
@@ -0,0 +1,15 @@
+.main-header {
+  top: 0;
+  left: 0;
+  width: 100%;
```
better to use 100vw (view width)

asafDovNetzer reviewed (changes requested):
overall really good

NoamGilad committed:
fixed
